src/sugar.jl

```diff
@@ -1162,31 +1162,31 @@ end
 
 
 """
-    seeded_autodiff_thunk(
+    autodiff(
         rmode::ReverseModeSplit,
-        dresult,
-        f,
-        ReturnActivity,
+        f::Annotation,
+        ReturnActivity::Type{<:Annotation},
+        dresult::Seed,
         annotated_args...
     )
     
-Call [`autodiff_thunk`](@ref), execute the forward pass, increment output tangent with `dresult`, then execute the reverse pass.
+Call [`autodiff_thunk`](@ref), execute the forward pass, increment output adjoint with `dresult`, then execute the reverse pass.
 
 Useful for computing pullbacks / VJPs for functions whose output is not a scalar.
 """
-function seeded_autodiff_thunk(
+function autodiff(
     rmode::ReverseModeSplit{ReturnPrimal},
-    dresult,
     f::FA,
     ::Type{RA},
+    dresult::Seed,
     args::Vararg{Annotation,N},
 ) where {ReturnPrimal,FA<:Annotation,RA<:Annotation,N}
     forward, reverse = autodiff_thunk(rmode, FA, RA, typeof.(args)...)
     tape, result, shadow_result = forward(f, args...)
     if RA <: Active
-        dinputs = only(reverse(f, args..., dresult, tape))
+        dinputs = only(reverse(f, args..., dresult.dval, tape))
     else
-        shadow_result .+= dresult  # TODO: generalize beyond arrays
+        shadow_result .+= dresult.dval  # TODO: generalize beyond arrays
```

Comment:
I think this can use `recursive_accumulate` nowadays.

Reply:
I used that function instead of `.+=` in the latest commit